addons/account/static/tests/grouped_tree_as_xslx_print_tests.js

```diff
@@ -0,0 +1,80 @@
+odoo.define('account.grouped_tree_print_as_xlsx', function (require) {
+"use strict";
+
+var AccountPrintXLSXListView = require('account.tree.print_xlsx');
+var testUtils = require('web.test_utils');
+
+var createView = testUtils.createView;
+var session = require('web.session');
+
+QUnit.module('Views', {
+    beforeEach: function () {
+        this.data = {
+            foo: {
+                fields: {
+                    foo: {string: "Foo", type: "char"},
+                    bar: {string: "Bar", type: "boolean"},
+                    date: {string: "Some Date", type: "date"}
+                },
+                records: [
+                    { id: 1, bar: true, foo: "yop", date: "2019-01-25" },
+                    { id: 2, bar: true, foo: "blip", date: false },
+                    { id: 3, bar: true, foo: "gnap", date: "2019-04-26" },
+                    { id: 4, bar: false, foo: "blip", date: "2019-07-24" }
+                ]
+            }
+        };
+    }
+}, function () {
+    QUnit.test('grouped list print as xlsx', async function (assert) {
+        assert.expect(7);
+
+        // save the session function
+        var oldGetFile = session.get_file;
+        session.get_file = function (option) {
+            var data = JSON.parse(option.data.data);
+
+            assert.deepEqual(data.columns, ['foo', 'bar', 'date'],
+                "columns sequence must be foo,bar,date");
+
+            var unFoldedRows = _.filter(data.rows, function(row) { 
+                return row.type == "list" && row.data.length == 0;
+            });
+            var foldedRows = _.filter(data.rows, function(row) { 
+                return row.type == "list" && row.data.length != 0;
+            });
+
+            assert.strictEqual(data.rows.length, 3, "there are total three rows");
+            assert.strictEqual(unFoldedRows.length, 2, "there are only two rows as grouped with no child record to each.");
+            assert.strictEqual(foldedRows.length, 1, "there are only one row with child records");
+
+            if (option.url == "/web/print_xlsx/export") {
+                assert.step("print_xlsx_rpc_call");
+            }
+            option.complete();
+            return Promise.resolve();
+        };
+
+        var list = await createView({
+            View: AccountPrintXLSXListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree js_class="account_print_as_xlsx">' +
+                    '<field name="foo"/>' +
+                    '<field name="bar"/>' +
+                    '<field name="date"/>' +
+                  '</tree>',
+            groupBy: ['foo']
+        });
+
+        // Opening first grouped list
+        await testUtils.dom.click(list.$(".o_group_header:eq(0)"));
+        assert.ok(list.$('.o_control_panel button.o_print_xslx').length, 'should have print button on control panel');
```

Comment:
assert.containsOnce is better, it checks that the element exists, is visible and only once there.